packages/happychat-connection/src/use-happychat-auth.ts

```diff
@@ -42,7 +44,7 @@ export async function requestHappyChatAuth() {
 }
 
 export default function useHappychatAuth( enabled = true ) {
-	return useQuery< HappychatAuth >( 'getHappychatAuth', requestHappyChatAuth, {
+	return useQuery< HappychatAuth >( 'getHappychatAuth' + key, requestHappyChatAuth, {
```

Comment:
Hi @alshakero :wave: I noticed this `Date.now()` key when reviewing the React Query upgrade in #74381 and I was wondering what's the purpose of this unique key. Every time this module is loaded, either on Calypso boot or when its async chunk is loaded, `key` will be initialized to a new unique value. What was wrong when the key was just `getHappychatAuth`?

The only difference I can see is persistence of the the query results to local storage. If you don't want to use the old persisted data after a Calypso reload, the unique `key` achieves that. But there's a much better way to do that, the `meta: { persist: false }` option that you can pass to `useQuery`.

Reply:
Hey! Yes, that was a bad hack on my part. I was completely baffled by the persistence because when I checked the RQ docs, persistence was off by default, but the response was being persisted, nonetheless, and it drove me mad. 

Especially the issue was intermittent (because this code in runs in the Gutenberg editor as well and there is no default persistence there) and I couldn't explain it, so I added the key. Only a month later or so, I learned we had a Calypso-level config that had persistence on and the `Date.now()` key was spamming the cache.

I just forgot to reverse this hack. I'll fix today.

Reply:
> when I checked the RQ docs, persistence was off by default

Yes, the `react-query` library doesn't do any persistence by default, the cache is only in memory. But in Calypso we're setting it up with the persistence plugin, and it's on by default for all queries, you need to opt-out manually with the `meta.persist: false` flag.

Reply:
If I remember correctly we would also need to fix the backend since it was serving cached response, and this hack was bypassing that.

Reply:
It's unlikely that the `key = Date.now()` hack would have any influence on backend. The query function doesn't use the `key` at all, it's not part of the REST request params.